### PR TITLE
refactor(Dates): save timestamps everywhere and use Date.parse

### PR DESCRIPTION
**Pull request by vaporox**
**Please describe the changes this PR makes and why it should be merged:**

- `Client#readyAt`, `GuildTemplate#createdAt` and `GuildTemplate#updatedAt` are now getters as opposed to their timestamp counterparts which are now properties
- Change from `new Date(x).getTime()` to `Date.parse(x)` where applicable for improved performance
- Fixed the timestamp property on the result of `MessageEmbed#toJSON` which was a `Date` instance and not a string
- Consistent implementation of date getters

**Status and versioning classification:**

- Code changes have been (partially) tested against the Discord API
- I know how to update typings and have done so

### CODE REVIEW

**Comment by iCrawl** [comment]
This needs a rebase.

File: src/structures/MessageEmbed.js [hunk 1]
```diff
@@ -488,7 +489,7 @@ class MessageEmbed {
       type: 'rich',
       description: this.description,
       url: this.url,
-      timestamp: this.timestamp && new Date(this.timestamp),
+      timestamp: this.createdAt?.toISOString(),
```
Comment (NotSugden) [inline src/structures/MessageEmbed.js:492]: this will be undefined instead of null now, probably not a big issue though

Comment (vaporox) [reply src/structures/MessageEmbed.js:492]: That's intended, https://discord.com/developers/docs/resources/channel#embed-object marks it as optional, not nullable

**Review by iCrawl — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Comment by Raiden-Ei** [comment]
After this PR, an error occurs when creating MessageEmbed: `TypeError: this.createdAt?.toISOString() is not a function`

**Pull request merged**